src/runner/mod.rs

```diff
@@ -96,6 +78,20 @@ impl Runner {
             }
         }
     }
+
+    fn prepare_before_run(&self) {
+        panic::set_hook(Box::new(|_panic_info| {
+            // XXX panics already catched at the test call site, don't output the trace in stdout
+        }));
+        let threads = if self.configuration.parallel { 0 } else { 1 }; // 0 is rayon default
+        let rayon_config = rayon::Configuration::new().num_threads(threads);
+        let _ = rayon::initialize(rayon_config);
```

Comment:
We might want to only call `rayon::initialize(…)` if `threads` is `1`, if it is we might want to check for the function's result and log an error message if appropriate?

```
/// If you do not call this function, the thread pool
/// will be automatically initialized with the default
/// configuration.

/// Initialization of the global thread pool happens exactly
/// once. Once started, the configuration cannot be
/// changed. Therefore, if you call `initialize` a second time, it
/// will return an error. An `Ok` result indicates that this
/// is the first initialization of the thread pool.
```
([Source](https://github.com/nikomatsakis/rayon/blob/248113e6f17dafbecc2c94b18997edb413579086/rayon-core/src/lib.rs#L326-L330))